### ISSUE TITLE
New eventually-consistent staging (for local dev and gno.land)

### ISSUE THREAD

**Issue by moul** [gnolang/gno#1131]
I suggest the creation of a special staging/testnet that resets regularly while retaining past transactions. After each reset, it will update (`git pull`), rebuild, and reload the `examples/` directory, attempting to replay previous raw transactions. If any issues arise, it will continue without interruption. This approach combines the advantages of a constantly refreshed test environment with the consistency of older ones.

This testnet not only provides an ideal space for iterative testing but also facilitates experimentation with chain upgradability. It serves as a playground for understanding and testing migrations, which is a crucial upcoming feature for our platform.

From the perspective of a contract developer, if my new concepts don't gain immediate traction, I can simply wait a day and try again. As long as the API of my previous contracts remains consistent, most transactions are likely to remain the same, even if there are changes behind the scenes. This can be particularly beneficial for features like our `r/gnoland/blog`, which is currently challenging to update and may encounter URL issues during potential migrations. I believe it can also serve as an excellent platform for a more dynamic `r/demo/boards`.

## To do

-  [x] **Naming:** Decide on a distinctive name for the new testnet. Consider rebranding or removing the current staging. -> gno.land
-  [ ] **History Storage:** Develop a fast method for storing and retrieving transaction histories.
-  [ ] **Script Updates:** Refine infrastructure scripts to cater to the unique needs of the new testnet.
-  [ ] **Tools to build valid genesis files** (#1189)

**Comment by albttx** [comment]
I like the idea! 

Can we move to a something like a semver for the chain, and deploy upgrades to every major/minor version?

For the nodes, I can manage an infra with

- 1 validator 
- 1-x RPCs
- 1 history node (with daily / weekly snapshots)

For the naming, I propose to keep the test{1...x} for other testnet and call this one testnet-x

**Comment by moul** [comment]
Let's adapt the existing staging.gno.land to make it eventually persistent.

**Comment by moul** [comment]
Bonuses:
-  Utilize https://github.com/gnolang/tx-exports for storing and restoring.
-  Subsequently, we will migrate this new testnet to "gno.land" (dependent on #1176).
-  Enable local execution of the "deploy a testnet with the latest git + try replaying history" functionality.

**Comment by moul** [comment]
Hey @zivkovicmilos, I've updated the original meta issue #1108 (portal loop) and created new related sub-issues:

-  #1187 (Gnoweb improvements) that can be tackled by anyone.
-  #1189 (Genesis manipulation libraries and tools) which could have been a part of your own issue, but I wanted to specify multiple things regarding gnosdk. It's a big enough task to have its own issue, but in my opinion, it's a sub-issue of your meta issue.

I apologize for any confusion, but the documentation is being progressively improved. Please feel free to continue. I have made the important bootstrap. Now, I suggest you continue drafting the plans with the team. Count on me for reviewing and providing feedback if you need it.

**Comment by zivkovicmilos** [comment]
@moul 
We have created a tool that aims to replace `gnotxsync`, with all the bells and whistles of `gnotxsync` and more:
https://github.com/gnolang/tx-archive

This tool also has the ability to support new sources of transactions (ex. the upcoming indexer), so it should be a good foundation. It falls under the `History Storage: Develop a fast method for storing and retrieving transaction histories.` in your top-level issue

**Comment by zivkovicmilos** [comment]
I have also created the issue that @ajnavarro has taken up in parallel for this effort, so we can track in the project board:
https://github.com/gnolang/gno/issues/1204

The idea is that we can use these Lego blocks to build out complicated functionality. The base functionality is:
- start node
- stop node
- restore txs
- backup txs

After #1203 and #1189 , the Lego blocks will also be:
- create genesis
- modify genesis

**Comment by moul** [comment]
By the way, @zivkovicmilos, I just recalled that I opened this issue on Hackerspace: https://github.com/gnolang/hackerspace/issues/23.
